Makefile

```diff
@@ -476,6 +476,10 @@ endif
 maybe-build-release:
 	./hack/maybe-build-release.sh
 
+release-notes: var-require-all-VERSION-GITHUB_TOKEN clean
+	@python3 -m pip install PyYAML PyGithub==2.3.0 && \
```

Comment:
I'm extremely wary about modifying the user's Python install without asking. IIUC this will also break on default installations of Ubuntu >= 23.10 due to the pip issue we've seen with Banzai.

We can create a temporary (?) venv to do this relatively simply, with something like the solution specified here:

https://stackoverflow.com/a/46188210

Reply:
Yeah, this is a no-go for me. 

We should really be using a venv or a docker container for this

Reply:
switched to a docker container